src/OrleansServiceBus/Providers/EventDataGeneratorStreamProvider/EventDataGeneratorStreamProvider.cs

```diff
@@ -87,12 +115,28 @@ public class AdapterFactory : EventHubAdapterFactory, IControllable
             /// <param name="svcProvider"></param>
             public override void Init(IProviderConfiguration providerCfg, string providerName, Logger log, IServiceProvider svcProvider)
             {
+                this.CheckpointerFactory = partition => NoOpCheckpointer.Create();
```

Comment:
Since NoOpCheckpointer is stateless, we should be able to use the same instance for all.  NoOpCheckpointer.Instance rather than NoOpCheckpointer.Create()?